libs/agno/agno/os/interfaces/agui/utils.py

```diff
@@ -334,6 +339,59 @@ def _create_events_from_chunk(
         step_finished_event = StepFinishedEvent(type=EventType.STEP_FINISHED, step_name="reasoning")
         events_to_emit.append(step_finished_event)
 
+    # Handle workflow-level events
+    elif chunk.event == WorkflowRunEvent.workflow_started:
+        workflow_name = getattr(chunk, "workflow_name", "workflow")
+        workflow_started_event = CustomEvent(
+            name="WorkflowStarted",
+            value={"workflow_name": workflow_name, "message": f"Starting workflow: {workflow_name}"},
+        )
+        events_to_emit.append(workflow_started_event)
+
+    elif chunk.event == WorkflowRunEvent.workflow_completed:
+        workflow_name = getattr(chunk, "workflow_name", "workflow")
+        workflow_completed_event = CustomEvent(
+            name="WorkflowCompleted",
+            value={"workflow_name": workflow_name, "message": f"Workflow completed: {workflow_name}"},
+        )
+        events_to_emit.append(workflow_completed_event)
+
+    elif chunk.event == WorkflowRunEvent.workflow_error:
+        error_message = getattr(chunk, "error", "Workflow error occurred")
+        workflow_name = getattr(chunk, "workflow_name", "workflow")
+        logger.error(f"Workflow error in {workflow_name}: {error_message}")
+        error_event = RunErrorEvent(type=EventType.RUN_ERROR, message=f"Workflow error: {error_message}")
+        events_to_emit.append(error_event)
+
+    elif chunk.event == WorkflowRunEvent.step_error:
+        error_message = getattr(chunk, "error", "Step error occurred")
+        step_name = getattr(chunk, "step_name", "unknown_step")
+        logger.error(f"Step error in {step_name}: {error_message}")
+        error_event = RunErrorEvent(type=EventType.RUN_ERROR, message=f"Step error in {step_name}: {error_message}")
+        events_to_emit.append(error_event)
+
+    # Handle workflow step events
+    elif chunk.event == WorkflowRunEvent.step_started:
```

Comment:
So similar to this, do we need to handle other events like `ConditionExecutionStartedEvent`, `ConditionExecutionCompletedEvent`, `RouterExecutionStartedEvent` ? 

What happens right now if we run a workflow those events coming in?